monitor/account/monitor.go

```diff
@@ -113,6 +134,97 @@ func monitorAccountOnce(
 	return nil
 }
 
+// monitorSolverNetRoleTokenOnce monitors solvernet role for provided token and chain.
+func monitorSolverNetRoleForever(
+	ctx context.Context,
+	network netconf.ID,
+	account eoa.Account,
+	backend *ethbackend.Backend,
+) {
+	// Not all network+chains+solver_roles need to be monitored
+	var found bool
+	for _, token := range eoa.SolverNetTokens() {
+		_, chainID := backend.Chain()
+		_, ok := eoa.GetSolverNetThreshold(account.Role, network, chainID, token)
+		if ok {
+			found = true
+		}
+	}
+	if !found {
+		return
+	}
+
+	ctx = log.WithCtx(ctx, "role", account.Role)
+	log.Info(ctx, "Monitoring solvernet role")
+
+	ticker := time.NewTicker(time.Second * 30)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			for _, token := range eoa.SolverNetTokens() {
+				loopCtx := log.WithCtx(ctx, "token", token)
+				err := monitorSolverNetRoleTokenOnce(loopCtx, network, account, backend, token)
+				if ctx.Err() != nil {
+					return
+				} else if err != nil {
+					log.Warn(loopCtx, "Monitoring solvernet role token failed (will retry)", err)
+				}
+			}
+		}
+	}
+}
+
+// monitorSolverNetRoleTokenOnce monitors solvernet role for provided token and chain.
+func monitorSolverNetRoleTokenOnce(
+	ctx context.Context,
+	network netconf.ID,
+	account eoa.Account,
+	backend *ethbackend.Backend,
+	token tokens.Token,
+) error {
+	// TODO(corver): improve this.
+	var stkn solverapp.Token
+	for _, tkn := range solverapp.AllTokens() {
+		if tkn.Token == token {
+			stkn = tkn
+		}
+	}
+	if stkn.Symbol == "" {
+		return errors.New("token not found", "token", token)
+	}
```

Comment:
will need to handle different chains. for ex, there are multiple sovler app tokens with `tkn.Token == tokens.WSTETH`

Reply:
yeah, I think this is supported